sp/src/game/server/ez2/npc_flyingpredator.cpp

```diff
@@ -474,13 +651,84 @@ int CNPC_FlyingPredator::SelectSchedule( void )
 			return SCHED_START_FLYING;
 		}
 
+		if ( CanUseFlyNav() && HasCondition( COND_HEAVY_DAMAGE ) )
+		{
+			m_flPlaybackRate = 1.25f;
+			return SCHED_TAKE_COVER_FROM_ENEMY;
+		}
+
 		break;
 
 	}
 
 	return BaseClass::SelectSchedule();
 }
 
+//=========================================================
+// GetSchedule 
+//=========================================================
+int CNPC_FlyingPredator::SelectFailSchedule( int failedSchedule, int failedTask, AI_TaskFailureCode_t taskFailCode )
+{
+	if (m_tFlyState == FlyState_Flying)
+	{
+		// Abort flight if we failed
+		SetFlyingState( FlyState_Falling );
+	}
+	else if (taskFailCode == FAIL_NO_ROUTE && m_tFlyState == FlyState_Walking && CanUseFlyNav())
+	{
+		// Try flying, that's a good trick
+		SetFlyingState( FlyState_Flying );
+	}
+
+	return BaseClass::SelectFailSchedule( failedSchedule, failedTask, taskFailCode );
+}
+
+//=========================================================
+// GetSchedule 
+//=========================================================
+void CNPC_FlyingPredator::BuildScheduleTestBits()
+{
+	BaseClass::BuildScheduleTestBits();
+
+	if ( IsFlying() )
+	{
+		SetCustomInterruptCondition( COND_FLY_STUCK );
+
+		if ( HasCondition( COND_CAN_RANGE_ATTACK1 ) && HasStrategySlotRange( SQUAD_SLOT_ATTACK1, SQUAD_SLOT_ATTACK2 ) )
+		{
+			if (IsCurSchedule( SCHED_APPROACH_ENEMY, false ) || IsCurSchedule( SCHED_CIRCLE_ENEMY, false ))
+			{
+				SetCustomInterruptCondition( COND_CAN_RANGE_ATTACK1 );
+			}
+		}
+	}
+	else if ( CapableOfFlyNav() && m_tFlyState == FlyState_Walking )
+	{
+		// Interrupt when we can fly
+		if ( IsCurSchedule( SCHED_CHASE_ENEMY, false ) )
+			SetCustomInterruptCondition( COND_CAN_FLY );
+	}
+}
+
+//=========================================================
+// GetSchedule 
+//=========================================================
+void CNPC_FlyingPredator::OnStartScene()
+{
+	BaseClass::OnStartScene();
+
+	if ( IsFlying() )
+	{
+		// HACKHACK: Slam velocity and pitch/roll to 0
```

Comment:
Why? What's the use case for this?

Reply:
Since most NPC functions only compensate for yaw and limited movetypes, this prevents flying stukabats from being caught at an incorrect pitch/roll or velocity when entering scripted sequences (i.e. dynamic interactions).

This is something which probably would've belonged in the PR for predator dynamic interaction fixes.